core/src/main/scala/com/pingcap/tispark/TiDBRelation.scala

```diff
@@ -37,18 +37,44 @@ case class TiDBRelation(session: TiSession,
   @transient val sqlContext: SQLContext
 ) extends BaseRelation
     with InsertableRelation {
-  val table: TiTableInfo = meta
-    .getTable(tableRef.databaseName, tableRef.tableName)
-    .getOrElse(
-      throw new TiClientInternalException(
-        "Table not exist " + tableRef + " valid databases are: " + meta.getDatabases
-          .map(_.getName)
-          .mkString("[", ",", "]")
-      )
-    )
+  lazy val table: TiTableInfo = getTableOrThrow(tableRef.databaseName, tableRef.tableName)
```

Comment:
recommend doing this (refactor or improvement) in another PR next time

Reply:
It is useful when TiFlash tables were missing from catalog. Without which we may not be able to determine the behavior of TiFlash replica.